main.tf

```diff
@@ -29,6 +29,23 @@ resource "azurerm_user_assigned_identity" "aks" {
   tags                = var.tags
 }
 
+data "azurerm_user_assigned_identity" "cluster_user_defined_identity" {
+  count = length(var.managed_identities.user_assigned_resource_ids) > 0 ? length(var.managed_identities.user_assigned_resource_ids) : 0
```

Comment:
```suggestion
  count = length(var.managed_identities.user_assigned_resource_ids) > 0 ? 1 : 0
```

Reply:
Did we talk about assumption that the identity is only 1 ?

Reply:
we did and decided for it to be only one, let me remove this